v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ReaderDataflowJobResourceManager.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.utils;
+
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatPipeline;
+
+import com.google.auth.Credentials;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineLauncher.LaunchInfo;
+import org.apache.beam.it.common.ResourceManager;
+import org.apache.beam.it.common.TestProperties;
+import org.apache.beam.it.common.utils.IORedirectUtil;
+import org.apache.beam.it.gcp.dataflow.FlexTemplateClient;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReaderDataflowJobResourceManager implements ResourceManager {
```

Comment:
Let's write a generic `DataflowResourceManager` with appropriate methods to execute any given Dataflow job? I think it will be a good addition to the framework. We may need as we build other components e.g DataValidator or MockDataGenerator Dataflow jobs.

This seems too specific to me.

Reply:
Done.